Core/Battler.gd

```diff
@@ -90,6 +90,10 @@ func _on_enemy_start_turn() -> void:
 
 
 func _handle_enemy_attack_queue() -> void:
+	# don't do anything if there are no attacks in queue
+	if _enemy_action_list.is_empty():
+		return
+		
```

Comment:
I get your point and think it would be better to do it your way

Should I remove my fix in `_handle_enemy_attack_queue`?
Since with the current implementation if there are enemies left, they will always add an attack into the `_enemy_action_list` making the check for empty list unnecessary

Reply:
mmh I mean we can keep it as a way to be sure, just in case
I would say remove it for now, implement the solution other solution, test that it fixes the bug then add your fix back. Better to have more checks than not enough, considering it won't have any cost on game performance

Reply:
Implemented the new solution and also kept the old solution